### PR TITLE
Add OCI Conformance validation with ORAS

### PR DESCRIPTION
**Pull request by SteveLasker**
Add several steps to validate if a registry supports OCI Artifact types using `oras push|pull` 

Signed-off-by: Steve Lasker <stevenlasker@hotmail.com>

### CODE REVIEW

File: docs/registry-conformance.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+- The `manifest.config` has something specific to the `x.sample` artifact. Since the object isn't of type `application/vnd.docker.container.image.v1+json`, all registry viewing tools should just ignore the content.
+- The `manifest.config.mediaType=application/x.sample`
+- The `layer.mediaType=application/txt`
+- Layer content = `artifact.txt`
+
+```bash
+oras push ${REGISTRY}/$REPO:artifact-config \
+  --manifest-config ./artifact-config.json:application/x.sample.v1+json \
+  ./artifact.txt:application/txt
```
Comment (sajayantony) [inline docs/registry-conformance.md:74]: would you like to use recommended mime types like text/plain? 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

**Review by sajayantony — CHANGES_REQUESTED** [review]

**Review by jdolitsky — CHANGES_REQUESTED** [review]
I think this is better to be codified. Perhaps a bash script that runs these commands, taking REGISTRY_URL REGISTRY_NAMESPACE REGISTRY_USERNAME REGISTRY_PASSWORD as env vars

**Comment by jdolitsky** [comment]
Can also wrap this in robot framework to get some better output. See this example in Helm: https://github.com/helm/acceptance-testing/blob/main/testsuites/repos.robot

**Comment by SteveLasker** [comment]
:eyes:
Thanks, folks. I'll get this incorporated next week.

**Comment by sajayantony** [comment]
This was a good starting point for the discussions that lead OCI artifacts. Given that we now have - https://github.com/opencontainers/distribution-spec/tree/main/conformance and ORAS repo has all the E2E tests planned do we need this PR or can we close it @SteveLasker

**Comment by SteveLasker** [comment]
This was initially created to help docker enable oci artifacts. Now that they’re done, every other registry now enables OcI Artifacts, and there’s better codified solution, happy to close this one.

**Pull request closed without merging**